src/main/java/org/folio/service/transformationfields/TransformationFieldsConfig.java

```diff
@@ -60,7 +64,9 @@
   PERMANENT_LOAN_TYPE_ID("permanentLoanTypeId", "$.item.permanentLoanTypeId", LOAN_TYPES),
   STATUS("status", "$.item.status.name"),
   VOLUME("volume", "$.item.volume"),
-  YEARCAPTION("yearCaption", "$.item.yearCaption");
+  YEARCAPTION("yearCaption", "$.item.yearCaption"),
+  ITEM_NOTE_TYPE("itemNoteTypeId", "$.item.notes[?(@.itemNoteTypeId=={id}) && ?(!(@.staffOnly) || @.staffOnly == false))]", ITEM_NOTE_TYPES),
```

Comment:
The id should be enclosed in quotes '' for it to work. Please check https://github.com/folio-org/mod-data-export/pull/147 or https://jsonpath.com/ for more details

Reply:
Also the path is incorrect, Please use https://jsonpath.com/ to validate your path

Reply:
I verified, works correctly without quotes

Reply:
https://github.com/json-path/JsonPath#filter-operators
> String literals must be enclosed by single or double quotes ([?(@.color == 'blue')] or [?(@.color == "blue")]).

Reply:
<img width="1680" alt="Screen Shot 2020-08-20 at 8 44 36 AM" src="https://user-images.githubusercontent.com/41596468/90772267-6e488280-e2c2-11ea-9bdc-25e291e3052c.png">
<img width="1676" alt="Screen Shot 2020-08-20 at 8 44 45 AM" src="https://user-images.githubusercontent.com/41596468/90772274-70aadc80-e2c2-11ea-92de-164bd341cde3.png">

Reply:
Atleast according to what I see quotes are needed. We can talk on a call if you findings are different

Reply:
@KVupp  you are right. It works without quotes only with simple text, without whitespaces and dashes. For UUIDs we need to go by using quotes.